### PR TITLE
add metrics support for CPU GPU peak memory usage

### PR DESCRIPTION
**Pull request by FindHao**
Add `--metrics [cpu_peak_mem,gpu_peak_mem,flops_dcgm]` to `run.py`.  Users could specify different combinations of those three metrics such as `--metrics cpu_peak_mem,gpu_peak_mem` to obtain the peak memory usage for CPU and GPU. The output is like the following.

```
GPU Peak Memory:                2.4268 GB
CPU Peak Memory:                4.1475 GB
```

### CODE REVIEW

File: components/model_analyzer/TorchBenchAnalyzer.py [hunk 1]
```diff
@@ -182,9 +232,36 @@ def calculate_flops(self, gpu_uuid=None) -> float:
...
+                gpu = self.gpu_factory.get_device_by_uuid(gpu_uuid)
+                return self.gpu_metric_value[gpu_uuid][GPUPeakMemory].value() / 1024
+            else:
+                raise TorchBenchAnalyzerException("No available GPU with uuid ", gpu_uuid, " found!")
+        if len(self.gpu_metric_value) > 1:
+            logger.warning("There are multiple available GPUs and will only return the first one's peak memory bandwidth.")
+        gpu_uuid = next(iter(self.gpu_metric_value))
+        return self.gpu_metric_value[gpu_uuid][GPUPeakMemory].value() / 1024
+
+    def calculate_cpu_peak_mem(self, cpu_uuid=None) -> float:
```
Comment (xuzhao9) [inline components/model_analyzer/TorchBenchAnalyzer.py:251]: We need CPU peak memory to work on CPU-only machines, i.e. a machine without Nvidia or DCGM

Comment (FindHao) [reply components/model_analyzer/TorchBenchAnalyzer.py:251]: Yes. The TorchBenchAnalzyzer works without GPU. It checks dcgm only when we add GPU-related metrics. It includes two monitors, cpu_monitor and gpu_monitor.

Comment (FindHao) [reply components/model_analyzer/TorchBenchAnalyzer.py:251]: Let me double check.

Comment (FindHao) [reply components/model_analyzer/TorchBenchAnalyzer.py:254]: forget to move the gpu information query from analyzer initialization to start_monitor. fixed by https://github.com/pytorch/benchmark/pull/1268/commits/0aefd63b22c96f6bc9810e515d224d123cf87094

**Pull request closed without merging**